src/units.jl

```diff
@@ -6,24 +6,35 @@ import ..DEFAULT_QUANTITY_TYPE
 
 @assert DEFAULT_VALUE_TYPE == Float64 "`units.jl` must be updated to support a different default value type."
 
-const _UNIT_SYMBOLS = Symbol[]
-const _UNIT_VALUES = DEFAULT_QUANTITY_TYPE[]
+const UNIT_SYMBOLS = Symbol[]
+const UNIT_VALUES = DEFAULT_QUANTITY_TYPE[]
 
 macro register_unit(name, value)
     return esc(_register_unit(name, value))
 end
 
+macro _lazy_register_unit(name, value)
+    return esc(_register_unit(name, value; lazy = true))
+end
+
 macro add_prefixes(base_unit, prefixes)
     @assert prefixes.head == :tuple
     return esc(_add_prefixes(base_unit, prefixes.args, _register_unit))
 end
 
-function _register_unit(name::Symbol, value)
-    s = string(name)
+const UNIT_MAPPING = Dict{Symbol,Int}()
+function update_unit_mapping(name, value, unit_mapping::Dict{Symbol, Int} = UNIT_MAPPING)
+    unit_mapping[name] = length(unit_mapping) + 1
+end
+
+function _register_unit(name::Symbol, value; lazy = false)
+    name_symbol = Meta.quot(name)
     return quote
+        haskey($UNIT_MAPPING, $name_symbol) && throw("Unit $name_symbol already exists.")
         const $name = $value
-        push!(_UNIT_SYMBOLS, Symbol($s))
-        push!(_UNIT_VALUES, $name)
+        push!($UNIT_SYMBOLS, $name_symbol)
+        push!($UNIT_VALUES, $name)
+        !$lazy && $update_unit_mapping($name_symbol, $value)
```

Comment:
Since all the calls are `_lazy` here, when does this function actually get called? (I assume it does, but I don't see where)

Reply:
Whenever externally a unit is registered, this executes and updates the UNIT_MAPPING; I'll extend it to update the `ALL_UNITS`, `ALL_MAPPING` and `ALL_SYMBOLS` too.

Reply:
Resolved with [`a4335c5` (#107)](https://github.com/SymbolicML/DynamicQuantities.jl/pull/107/commits/a4335c5e5792d947932766b62eb4f4e836437d29)